### PR TITLE
Re-create `templates/app`, add deprecation notices for `app_shared` and `skeleton`.

### PR DESCRIPTION
**Pull request by matanlurey**
Part of https://github.com/flutter/flutter/issues/160673, which removes `--template skeleton`.

This PR copies the `templates/app_shared` directory, which was only used to share code between `templates/app` and `templates/skeleton`, in preparation for removing `skeleton` (and making the existence of `app_shared` obsolete).

Issue tracking final cleanup: https://github.com/flutter/flutter/issues/160692.

### CODE REVIEW

**Comment by ditman** [comment]
We need to update the CHANGELOG and pubspec.yaml to have this autopublished in pub.dev, so it can be consumed from the flutter tool.

The "Linux analyze" tests are failing because two new dart lints.

(I'll publish a temporary ignore for those to reopen the tree)

**Review by ditman — APPROVED** [review]

**Comment by ditman** [comment]
Working on the analysis failures here:

https://github.com/flutter/flutter/issues/160799

Will land ASAP.

File: packages/flutter_template_images/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ## NEXT
```
Comment (stuartmorgan) [inline packages/flutter_template_images/CHANGELOG.md:1]: Without a version bump this won't be usable.

**Review by stuartmorgan — DISMISSED** [review]

File: packages/flutter_template_images/CHANGELOG.md (hunk 1 above)
Comment (matanlurey) [reply packages/flutter_template_images/CHANGELOG.md:1]: Done!

**Pull request merged**